query/security/ldap/src/test/groovy/org/codice/ddf/admin/ldap/discover/LdapTestBindSpec.groovy

```diff
@@ -0,0 +1,228 @@
+import org.codice.ddf.admin.ldap.fields.connection.*
+import spock.lang.Ignore
+import spock.lang.Specification
+
+import static org.codice.ddf.admin.ldap.LdapTestingCommons.noEncryptionLdapConnectionInfo
+import static org.codice.ddf.admin.ldap.LdapTestingCommons.simpleBindInfo
+
+class LdapTestBindSpec extends Specification {
+    static TestLdapServer server
+    Map<String, Object> args
+    LdapTestBind ldapBindFunction
+
+    def setupSpec() {
+        server = TestLdapServer.getInstance().useSimpleAuth()
+        server.startListening()
+    }
+
+    def cleanupSpec() {
+        server.shutdown()
+        server = null
+    }
+
+    def setup() {
+        LdapTestingCommons.loadLdapTestProperties()
+        ldapBindFunction = new LdapTestBind()
+    }
+
+    def 'Fail on missing required fields'() {
+        setup:
+        def baseMsg = [LdapTestBind.FIELD_NAME, FunctionField.ARGUMENT]
+        def missingHostMsgPath = baseMsg + [LdapConnectionField.DEFAULT_FIELD_NAME, HostnameField.DEFAULT_FIELD_NAME]
+        def missingPortMsgPath = baseMsg + [LdapConnectionField.DEFAULT_FIELD_NAME, PortField.DEFAULT_FIELD_NAME]
+        def missingEncryptMsgPath = baseMsg + [LdapConnectionField.DEFAULT_FIELD_NAME, LdapEncryptionMethodField.DEFAULT_FIELD_NAME]
+        def missingUsernameMsgPath = baseMsg + [LdapBindUserInfo.DEFAULT_FIELD_NAME, CredentialsField.DEFAULT_FIELD_NAME, CredentialsField.USERNAME_FIELD_NAME]
+        def missingUserpasswordMsgPath = baseMsg + [LdapBindUserInfo.DEFAULT_FIELD_NAME, CredentialsField.DEFAULT_FIELD_NAME, CredentialsField.PASSWORD_FIELD_NAME]
+        def missingBindMethodMsgPath = baseMsg + [LdapBindUserInfo.DEFAULT_FIELD_NAME, LdapBindMethod.DEFAULT_FIELD_NAME]
+
+        when:
+        FunctionReport report = ldapBindFunction.getValue()
+
+        then:
+        report.messages().size() == 6
+        report.messages().count {
+            it.getCode() == DefaultMessages.MISSING_REQUIRED_FIELD
+        } == 6
+
+        report.messages()*.getPath() as Set == [missingHostMsgPath, missingPortMsgPath, missingEncryptMsgPath,
+                                                missingUsernameMsgPath, missingUserpasswordMsgPath, missingBindMethodMsgPath] as Set
+    }
+
+    def 'Fail on missing bind realm'() {
+        setup:
+        args = [(LdapBindUserInfo.DEFAULT_FIELD_NAME)   : digestBindInfo().realm(null).getValue(),
+                (LdapConnectionField.DEFAULT_FIELD_NAME): noEncryptionLdapConnectionInfo().getValue()]
+        ldapBindFunction.setValue(args)
+
+        when:
+        FunctionReport report = ldapBindFunction.getValue()
+
+        then:
+        report.messages().size() == 1
+        report.messages().get(0).getCode() == DefaultMessages.MISSING_REQUIRED_FIELD
+        report.messages().get(0).getPath() == [LdapTestBind.FIELD_NAME, FunctionField.ARGUMENT, LdapBindUserInfo.DEFAULT_FIELD_NAME, LdapRealm.DEFAULT_FIELD_NAME]
+    }
+
+    def 'Fail to connect to LDAP'() {
```

Comment:
We could/should explore extracting out these commons tests between these classes and putting them in a super class and having these tests extend the super class.